nix/haskell-packages.nix

```diff
@@ -0,0 +1,126 @@
+{ compiler, enableDhall, enableSwagger }:
+
+pkgsNew: pkgsOld:
+
+let
+  gitignoreSource =
+    let
+      source = pkgsNew.fetchFromGitHub {
+        owner = "hercules-ci";
+        repo = "gitignore.nix";
+        rev = "211907489e9f198594c0eb0ca9256a1949c9d412";
+        sha256 = "06j7wpvj54khw0z10fjyi31kpafkr6hi1k0di13k1xp8kywvfyx8";
+      };
+    in (import source { inherit (pkgsNew) lib; }).gitignoreSource;
+
+in {
+  haskellPackages = pkgsOld.haskell.packages."${compiler}".override (old: {
+    overrides =
+      pkgsOld.lib.composeExtensions
+        (old.overrides or (_: _: {}))
+        (haskellPackagesNew: haskellPackagesOld:
+          let
+            proto3-suite-flags = (if enableDhall then ["-fdhall"] else []) ++ (if enableSwagger then [] else ["-f-swagger"]);
+
+          in {
+            tasty = haskellPackagesNew.callPackage ./haskell-packages/tasty.nix { };
+
+            tasty-ant-xml = haskellPackagesNew.callPackage ./haskell-packages/tasty-ant-xml.nix { };
+
+            proto3-wire =
+              let
+                source = pkgsNew.fetchFromGitHub {
+                  owner = "awakesecurity";
+                  repo = "proto3-wire";
+                  rev = "e5e0158ceaaa50d258bb86dbf2e6c42d5e16c3c5";
+                  sha256 = "14r2qm6x4bcaywbi3cypriz4hr8i2v3j4qm61lal6x21p0z9i9ak";
+                };
+              in haskellPackagesNew.callCabal2nix "proto3-wire" source { };
+
+            proto3-suite-base =
+              haskellPackagesNew.callCabal2nixWithOptions
+                "proto3-suite"
+                (gitignoreSource ../.)
+                (pkgsNew.lib.concatStringsSep " " proto3-suite-flags)
+                { };
```

Comment:
Why remove the `pname` and `configureFlags` overrides for `proto3-suite-base`?

I definitely want to keep the `pname` override. The `configureFlags` might not be necessary, but I'm not sure; was that something you tested?

If they aren't necessary, then I think you should inline `proto3-suite-flags` since it's only being used in one place.

Reply:
The removal of `pname` was not intentional but I did remove `configureFlags` in the attribute overrides since it is redundant.